cloudify_agent/api/pm/initd.py

```diff
@@ -280,6 +286,27 @@ def _create_config(self):
         self.runner.sudo('cp {0} {1}'.format(rendered, self.config_path))
         self.runner.sudo('rm {0}'.format(rendered))
 
+    def _create_start_on_boot_entry(self):
+
+        response = self.runner.run('which dpkg', exit_on_failure=False)
+        if response.code == 0:
+            # debian based distribution, run update-rc.d
+            self.runner.sudo('update-rc.d {0} defaults'.format(self.name))
+        else:
+            response = self.runner.run('which rpm')
+            if response.code == 0:
+                # rpm based distribution, run chkconfig
+                self.runner.sudo('/sbin/chkconfig --add {0}'
+                                 .format(self.name))
+                self.runner.sudo('/sbin/chkconfig {0} on'
+                                 .format(self.name))
+            else:
+                raise errors.DaemonConfigurationError(
+                    "Cannot create a start-on-boot entry. Unknown "
+                    "distribution base. Supported distributions bases are "
+                    "debian and RPM"
```

Comment:
why? aren't there other distros that use RPM that are not RHEL where this can work on? openSuse, fedora...